website/docs/setup/02_server.mdx

```diff
@@ -1,23 +1,97 @@
 ---
-title: KitQL - GraphQL Server
-sidebar_label: GraphQL Server
+title: KitQL
+sidebar_label: KitQL
 ---
 
-# ⚡GraphQL Server
+# ⚡KitQL
 
-This set of packages will take care of the backend part of GraphQL in SvelteKit.
-
-We will use:
+We make use of amazing Open Source projects:
 
 - [Yoga](https://www.graphql-yoga.com/)
-- [Envelop](https://www.envelop.dev/)
 - [Code Generator](https://www.graphql-code-generator.com/)
 - [Modules](https://www.graphql-modules.com/)
+- [Envelop](https://www.envelop.dev/)
 
 ## Configuration
 
 <br />
-### Code Generator
+### Hooks
+
+The idea is to bring functionalities with hooks, something like:
+```ts
+// hooks.server.ts
+import { dev } from '$app/environment'
+import { kitqlServer } from '$lib/graphql/kitQLServer'
+import { graphiql, graphql } from '@kitql/all-in'
+import { sequence } from '@sveltejs/kit/hooks'
+
+export const handle = sequence(
+  // Add graphql
+  graphql(kitqlServer),
+
+  // graphiql
+  graphiql({
+    enabled: dev,
+  })
+)
+```
+
+_You could use only `graphiql` here for example!_
+
+Now, let's build the `kitqlServer` instance 👇
+
+<br />
+### kitqlServer
+
+This is the file where you will create your kitQL Server. To start, it should look like:
+
+```ts
+import { createServer } from '@graphql-yoga/common'
+import type { RequestEvent } from '@sveltejs/kit'
+import { kitQLModules } from '@kitql/all-in'
+// This should be fully generated 🥳
+import { modules } from './_kitql/_appModules'
+
+// export type IKitQLContext ables to use it in code gen, and have resolvers fully typed!
+// config:
+//   contextType: $graphql/kitQLServer#IKitQLContext
+export type IKitQLContext = {
+  request: Request
+  user?: {
+    id: number
+    name: string
+  }
+}
+
+function getContext({ request }: RequestEvent) {
```

Comment:
How does this context interact with the houdini session?

Reply:
Hoo, this has nothing to do.
It's to build the 3rd argument in resolver
```ts
export const resolvers: _InitModule.Resolvers = {
  Mutation: {
    _boostServer: async (root, args, ctx) => {
      // ctx                    <------ This context (out of the 20 000 context that we have ^^)
      return `Done! KitQL is on 🔥`
    },
  },
}
```

Reply:
Ohh gotcha! Do you think there could be value in connecting them? i'm not sure about the exact details but it seems like it might be interesting to give users a consistent way to talk about the current user across the whole stack

Reply:
Yeah, not sure exactly how yet, but I'm sure we could have an `authHandle` doing the right thing on both sides. 👍👍👍